flink-core/src/main/java/org/apache/flink/api/common/eventtime/WatermarkOutputMultiplexer.java

```diff
@@ -86,6 +86,7 @@ public void registerNewOutput(String id) {
         checkState(previouslyRegistered == null, "Already contains an output for ID %s", id);
 
         combinedWatermarkStatus.add(outputState);
+        underlyingOutput.emitWatermark(Watermark.MIN_WATERMARK);
```

Comment:
Is this also necessary if there has been a split already?

Reply:
You are right, it won't work that way. I took a different approach.